src/server/page_server.hpp

```diff
@@ -179,7 +179,54 @@ class page_server
 	std::mutex m_abnormal_tran_server_disconnect_mtx;
 	bool m_abnormal_tran_server_disconnect;
     };
-    using connection_handler_uptr_t = std::unique_ptr<connection_handler>;
+
+    /*
+     *  TODO add some explanation and diagrams for this.
+     */
+    class follower_connection_handler
+    {
+      public:
+	using follower_server_conn_t =
+		cubcomm::request_sync_client_server<followee_to_follower_request, follower_to_followee_request, std::string>;
+
+	follower_connection_handler () = delete;
+	follower_connection_handler (cubcomm::channel &&chn, page_server &ps);
+
+	follower_connection_handler (const connection_handler &) = delete;
+	follower_connection_handler (connection_handler &&) = delete;
+
+	follower_connection_handler &operator= (const connection_handler &) = delete;
+	follower_connection_handler &operator= (connection_handler &&) = delete;
+
+      private:
+	void receive_dummy_request (follower_server_conn_t::sequenced_payload &&a_sp);  // TODO remove it
+
+	page_server &m_ps;
+	std::unique_ptr<follower_server_conn_t> m_conn;
+    };
+
+    class followee_connection_handler
+    {
+      public:
+	using followee_server_conn_t =
```

Comment:
`followee_server_conn_t` can be private; same elsewhere

Reply:
Thanks. This will be fixed in the following PR.